uportal-portlets-overlay/cas-proxy-test-portlet/src/main/java/org/jasig/portlet/cas/test/mvc/ProxyCasController.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.portlet.cas.test.mvc;
+
+import javax.portlet.PortletRequest;
+import javax.portlet.RenderRequest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jasig.cas.client.validation.Assertion;
+import org.jasig.cas.client.validation.TicketValidationException;
+import org.jasig.cas.client.validation.TicketValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.portlet.ModelAndView;
+
+/**
+ *
+ *
+ * @author Jen Bourey, jbourey@unicon.net
+ * @version $Revision$
+ */
+@Controller
+@RequestMapping("VIEW")
+public final class ProxyCasController {
+
+	private Log log = LogFactory.getLog(ProxyCasController.class);
+
+	private TicketValidator validator;
+
+	/**
+	 * Set the ticket validator to use for proxy ticket validation.
+	 *
+	 * @param validator
+	 */
+	@Autowired(required = true)
+	public void setTicketValidator(TicketValidator validator) {
+		this.validator = validator;
+	}
+
+	private String serviceUrl = "http://localhost:8080/cas-proxy-test-portlet";
+
+	public void setServiceUrl(String serviceUrl) {
+		this.serviceUrl = serviceUrl;
+	}
+
+	private String proxyTicketKey = "casProxyTicket";
+
```

Comment:
Since this does not have a setter, it should be `private static final`.
This should also be moved up.